### PR TITLE
feat: Add release testing workflow for public networks

### PR DESCRIPTION
**Pull request by beeradb**
This PR refactors the public acceptance workflow to do the following:

- Adds support for mainnet
- Run only the "release_tests" tag against public networks.
- Use the new `TEST_INITIAL_ACCOUNT_STARTING_BALANCE` attribute to ensure tests will allow runs with only 500 hbar in the operator account.
- Provide configurations for mainnet environment.

Fixes: #2947

### CODE REVIEW

File: packages/server/tests/mainnetAcceptance.env [hunk 1]
```diff
@@ -0,0 +1,7 @@
+HEDERA_NETWORK=mainnet
+CHAIN_ID=0x00
```
Comment (quiet-node) [inline packages/server/tests/mainnetAcceptance.env:2]: should we match the mainnet chain ID 0x127?

**Review by ebadiere — COMMENTED** [review]
LG, but Logan's question about chainId should be resolved.

File: packages/server/tests/mainnetAcceptance.env (hunk 1 above)
Comment (beeradb) [reply packages/server/tests/mainnetAcceptance.env]: good catch - fixed

**Review by ebadiere — APPROVED** [review]
LG.

**Review by quiet-node — APPROVED** [review]

**Pull request merged**